images/ubuntu/scripts/build/install-docker.sh

```diff
@@ -14,33 +14,36 @@ REPO_PATH="/etc/apt/sources.list.d/docker.list"
 os_codename=$(lsb_release -cs)
 
 curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o $GPG_KEY
-echo "deb [arch=amd64 signed-by=$GPG_KEY] $REPO_URL ${os_codename} stable" > $REPO_PATH
+echo "deb [arch=amd64 signed-by=$GPG_KEY] $REPO_URL ${os_codename} stable" >$REPO_PATH
 apt-get update
 
-for pkg in containerd.io docker-ce-cli docker-ce docker-buildx-plugin; do
-    version=$(get_toolset_value ".docker.components.\"$pkg\"")
+# Install docker components which available via apt-get
+# Using toolsets keep installation order to install dependencies before the package in order to control versions
+
+components=$(get_toolset_value '.docker.components[] .package')
+for package in $components; do
+    version=$(get_toolset_value ".docker.components[] | select(.package == \"$package\") | .version")
     if [[ $version == "latest" ]]; then
-        components_to_install+="${pkg} "
+        apt-get install -y --no-install-recommends "$package"
     else
-        version_string=$(apt-cache madison "${pkg}" | awk '{ print $3 }' | grep "${version}" | grep "${os_codename}" | head -1)
-        components_to_install+="${pkg}=${version_string} "
+        version_string=$(apt-cache madison "$package" | awk '{ print $3 }' | grep "$version" | grep "$os_codename" | head -1)
+        apt-get install -y --no-install-recommends "${package}=${version_string}"
     fi
 done
-apt-get install -y --no-install-recommends $components_to_install
-
-# Download docker compose v2 from releases
-# Temporaty pinned to v2.23.3 due https://github.com/actions/runner-images/issues/9172
-compose_version=$(get_toolset_value ".docker.components.compose")
-URL=$(resolve_github_release_asset_url "docker/compose" "endswith(\"compose-linux-x86_64\")" "${compose_version}")
-compose_binary_path=$(download_with_retry "${URL}" "/tmp/docker-compose-v2")
 
-# Supply chain security - Docker Compose v2
-compose_hash_url=$(resolve_github_release_asset_url "docker/compose" "endswith(\"checksums.txt\")" "${compose_version}")
```

Comment:
wait, we're removing this?

Reply:
this was the old style of installing this, if you look at the HEAD of runner-images they now install it using plugins like we do